generate_report.py

```diff
@@ -0,0 +1,27 @@
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+from xmodule.modulestore.django import modulestore
+from helpers.classes import Course
+from helpers.helper import (
+    populate_block_heirarchy,
+    generate_html_report
+)
+
+def generate_report():
+    active_courses = []
+    store = modulestore()
+
+    print("Fetching list of all courses")
+    active_course_overviews = CourseOverview.get_all_courses(active_only=True)
```

Comment:
```suggestion
    course_overviews = CourseOverview.get_all_courses()
    # This can be replaced with `CourseOverview.get_all_courses(active_only=True)` in Palm.
    active_course_overviews = course_overviews.filter(
        Q(end__isnull=True) | Q(end__gte=datetime.now().replace(tzinfo=pytz.UTC))
    )
    
```

`active_only` was introduced in Palm. We'll also need the following imports:
```python
from datetime import datetime
from django.db.models import Q
import pytz
```

Reply:
Changed